okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/HostMetricsRegistry.java

```diff
@@ -33,33 +33,38 @@
 
     private static final Logger log = LoggerFactory.getLogger(HostMetricsRegistry.class);
 
-    private final LoadingCache<ServiceAndHost, DefaultHostMetrics> hostMetrics;
+    private final LoadingCache<ServiceHostAndPort, DefaultHostMetrics> hostMetrics;
 
     HostMetricsRegistry() {
         this.hostMetrics = CacheBuilder.newBuilder()
                 .maximumSize(1_000)
                 .expireAfterAccess(1, TimeUnit.DAYS)
-                .build(new CacheLoader<ServiceAndHost, DefaultHostMetrics>() {
+                .build(new CacheLoader<ServiceHostAndPort, DefaultHostMetrics>() {
                     @Override
-                    public DefaultHostMetrics load(ServiceAndHost key) {
-                        return new DefaultHostMetrics(key.serviceName(), key.hostname(), Clock.systemUTC());
+                    public DefaultHostMetrics load(ServiceHostAndPort key) {
+                        return new DefaultHostMetrics(key.serviceName(), key.hostname(), key.port(), Clock.systemUTC());
                     }
                 });
     }
 
-    void record(String serviceName, String hostname, int statusCode, long micros) {
+    void record(String serviceName, String hostname, int port, int statusCode, long micros) {
         try {
-            hostMetrics.getUnchecked(ImmutableServiceAndHost.of(serviceName, hostname)).record(statusCode, micros);
+            hostMetrics.getUnchecked(
+                    ImmutableServiceHostAndPort.of(serviceName, hostname, port)).record(statusCode, micros);
         } catch (Exception e) {
-            log.warn("Unable to record metrics for host", UnsafeArg.of("hostname", hostname));
+            log.warn("Unable to record metrics for host and url",
+                    UnsafeArg.of("hostname", hostname),
+                    UnsafeArg.of("url", port));
```

Comment:
first arg should be `port`

Reply:
port is always safe btw, it's a plain number.

Reply:
whoops, fail on my part to rename.